### PR TITLE
[4.0] More cleanup on the jstools

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- Remove the .eslintignore
- Fix the eslint errors on the remaining WebComponents
- Add a try/catch for any Json.parse()
- Favour functional forEach over for loops

### Testing Instructions
All fields are working as before

### CODE REVIEW

**Comment by dgrammatiko** [comment]
Hey @wilsonge if you get some free time can you check this one? Thanks

File: .eslintignore [hunk 1]
```diff
@@ -1,15 +0,0 @@
-# A list of files to ignore from linting
-*.js
-!*.es6.js
-administrator/components/com_media/node_modules/uri-js/node_modules/punycode/punycode.es6.js
-media
```
Comment (wilsonge) [inline .eslintignore:5]: I think we at least want to keep the media directory in here - as that will contain the minifed/vendor files etc.

Comment (wilsonge) [reply .eslintignore:5]: Ignore. I just reached the bottom of the diff

**Comment by wilsonge** [comment]
LGTM! Thanks!

**Comment by Quy** [comment]
@dgrammatiko Can you please look into this? Thanks.

To reproduce, go to Global Configuration > Users > Email Domain Options and add a row.

```
Uncaught TypeError: rows.forEach is not a function
    getRows http://localhost/joomla-cms-4.0-dev/media/system/js/fields/joomla-field-subform.js:150
    JoomlaFieldSubform http://localhost/joomla-cms-4.0-dev/media/system/js/fields/joomla-field-subform.js:84
    <anonymous> http://localhost/joomla-cms-4.0-dev/media/system/js/fields/joomla-field-subform.js:585
    <anonymous> http://localhost/joomla-cms-4.0-dev/media/system/js/fields/joomla-field-subform.js:586c
```

**Comment by dgrammatiko** [comment]
@Quy thanks for catching this, check #31213

**Pull request merged**